webpush/encrypt.go

```diff
@@ -139,36 +142,64 @@ type EncryptionResult struct {
 	ServerPublicKey []byte
 }
 
+// Decrypt an encrypted messages.
```

Comment:
The UA use case feels like a step too far for the core goal of this library. Is there a way to keep this for the sending use case and a seperate module could inherit this module and add on the UA specific bits?